src/kv/map.h

```diff
@@ -2,723 +2,202 @@
 // Licensed under the Apache 2.0 License.
 #pragma once
 
-#include "ds/dl_list.h"
-#include "ds/logger.h"
-#include "ds/spin_lock.h"
-#include "kv_serialiser.h"
 #include "kv_types.h"
+#include "serialise_entry_json.h"
+#include "serialise_entry_msgpack.h"
 #include "tx_view.h"
 
-#include <functional>
-#include <optional>
-#include <unordered_set>
-
 namespace kv
 {
-  namespace Check
-  {
-    struct No
-    {};
-
-    template <typename T, typename Arg>
-    No operator!=(const T&, const Arg&)
-    {
-      return No();
-    }
-
-    template <typename T, typename Arg = T>
-    struct Ne
-    {
-      enum
-      {
-        value = !std::is_same<decltype(*(T*)(0) != *(Arg*)(0)), No>::value
-      };
-    };
-
-    template <class T>
-    bool ne(std::enable_if_t<Ne<T>::value, const T&> a, const T& b)
-    {
-      return a != b;
-    }
-
-    template <class T>
-    bool ne(std::enable_if_t<!Ne<T>::value, const T&> a, const T& b)
-    {
-      return false;
-    }
-  }
-
-  template <class K, class V, class H = std::hash<K>>
-  class Map;
-
-  template <class K, class V, class H>
-  class TxViewCommitter : public AbstractTxView
+  template <typename K, typename V, typename KSerialiser, typename VSerialiser>
+  class TypedMap : public AbstractMap
```

Comment:
This was `kv::experimental::Map`, now its `kv::TypedMap`, providing the `kv::Map` replacement via the typedefs at the bottom of this file.